Automattic/wp-calypso
# BackupBrowser: Add file details scaffolding and row counts for database tables

We are currently working on the API to fetch files info when clicking on them while navigating on the backup browser. In the meantime, we are including the number of rows when clicking a database table (`/sql/*`) as we don't require an additional API to get that information.

## Proposed Changes

* Add `FileInfoCard` component, which will be in charge of rendering the file details depending on the item type. For now, it only works for database tables.
* Add the number of rows backed up when a user clicks on a database table.
* Refactor the way we handle clicks on items. Now all items could be opened, not just the ones with `hasChildren === true`, but only those will render children nodes.
* Add a `activeNodePath` state in `FileBrowser` so we could track the current node opened. This will help us to expand the file details on one node when clicked and collapse the previous opened.

## Screenshot

<img width="622" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/e6801dc1-b32e-4dc6-bbbf-03e5bb694f34">



## Testing Instructions



* Start a Jetpack Cloud or Calypso Live branch
* Select one of your sites with a Jetpack VaultPress Backup plan.
* Navigate to VaultPress Backup page
* Pick one date with a backup available (use the calendar and click on any date marked with a dot).
* Browse the `/sql/` directory and clicks on any database table. It should open a card with the number of rows that table includes.
* Browse other directories and ensure clicking on an item doesn't display the info card.

## Pre-merge Checklist
